cpp/src/arrow/engine/substrait/relation_internal.h

```diff
@@ -30,6 +30,12 @@
 #include "substrait/algebra.pb.h"  // IWYU pragma: export
 
 namespace arrow {
+namespace compute {
```

Comment:
Why do we need this here?

Reply:
We need a forward declaration of `Aggregate` for the signature of `ParseAggregateMeasure`. However, we no longer need `AggregateNodeOptions`, so I'll remove it.

Reply:
I see. I believe if we don't expose these two methods from relation_internal.h then this we don't need this fwd declaration either. See https://github.com/apache/arrow/pull/34627#discussion_r1151928873

Reply:
I don't see now how we can cleanly avoid exposing the two functions in the "internal"-namespace without putting the function calling them in an inappropriate place. If we can do this cleanly, then I agree.

Reply:
Sth like this - https://github.com/apache/arrow/commit/e958b6d6cfd8889fd8e5a616eadfe0023cac3b77

I moved all things under relation_internal.cc which I think makes the header file cleaner.

The downside is that the extension logic is now inside relation_internal.cc which makes that file larger, but I still think this is justifiable because "aggregate" and "segment aggregate" does share internal logic here. This is cleaner than exposing the `ParseAggregateMeasure` and `MakeAggregateDeclaration` in `relation_internal.h`.

Reply:
I understand what you're trying to achieve, however, I think this refactoring of `options.cc` should be out of scope for this PR and in any case should probably be into new files rather than into `relation_internal.cc`, which can't be the right place.